### PR TITLE
Running dataclass

### PR DESCRIPTION
**Pull request by Siupahc**
# Feature

- New RunningWorkout class, heriting from Workout

# Implementation details

- There's 2 new parameters, time and path, path is a list of list of [LatLng](https://developers.google.com/android/reference/com/google/android/gms/maps/model/LatLng) and time is a Long containing the duration in milliseconds. I spent litteral hours trying to serialize the [Duration](https://kotlinlang.org/api/core/kotlin-stdlib/kotlin.time/-duration/l) type, and I had non-sense exceptions, so a Long should be good

### CODE REVIEW

**Comment by Siupahc** [comment]
Also, the WorkoutViewModel + Repo should work with the Running, so no need for new one

**Comment by Siupahc** [comment]
> Also, the WorkoutViewModel + Repo should work with the Running, so no need for new one

Spoiler : it didn't

File: app/src/main/java/com/android/sample/model/workout/WorkoutRepositoryFirestore.kt [hunk 1]
```diff
@@ -275,4 +277,41 @@ open class WorkoutRepositoryFirestore<T : Workout>(
...
+    @FromJson
+    fun fromJson(jsonList: List<Map<String, Double>>): List<LatLng> {
+      return jsonList.map { json ->
+        LatLng.newBuilder()
+            .setLatitude(json["latitude"] ?: throw JsonDataException("Missing latitude"))
+            .setLongitude(json["longitude"] ?: throw JsonDataException("Missing longitude"))
+            .build()
+      }
+    }
+  }
```
Comment (didi2101) [inline app/src/main/java/com/android/sample/model/workout/WorkoutRepositoryFirestore.kt:316]: Really clear !

File: app/src/androidTest/java/com/android/sample/ui/workout/WorkoutScreenTest.kt [hunk 2]
```diff
@@ -151,7 +153,6 @@ class WorkoutScreenTest {
               url = "sampleUrl")
         }
 
-    // Mock the videos flow
     whenever(mockVideoRepository.getVideos(anyOrNull(), anyOrNull())).thenAnswer {
       val onSuccess = it.getArgument<Function1<List<Video>, Unit>>(0)
       onSuccess(sampleVideos)
```
Comment (didi2101) [inline app/src/androidTest/java/com/android/sample/ui/workout/WorkoutScreenTest.kt]: I don't get it why it is in this PR..

File: app/src/main/java/com/android/sample/model/workout/RunningWorkout.kt [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    userIdSet: MutableSet<String> = mutableSetOf(),
+    exercises: MutableList<Exercise> = mutableListOf(),
+    date: LocalDateTime,
+    @Json(name = "path") val path: List<LatLng>,
+    @Json(name = "time") val timeMs: Long
+) : Workout(workoutId, name, description, false, userIdSet, mutableListOf(), date) {
+  companion object {
+    const val DOCUMENT_NAME = "runningWorkout"
+  }
+}
```
Comment (didi2101) [inline app/src/main/java/com/android/sample/model/workout/RunningWorkout.kt:34]: Simple and concise way of implementing this class.

**Review by didi2101 — APPROVED** [review]
The job is really good, the feature is well implemented, nothing to say.

Approved.

**Pull request merged**